### PR TITLE
Update schema to add support for kernel and discriminator selection to QASM

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

This PR updates the Qobj-QASM config schema for kernel and discriminator selection to Qiskit measure command

### CODE REVIEW

**Comment by taalexander** [comment]
What should we do about the version number here @mtreinish? Should we merge this and sort out the related issue in the future?

**Comment by mtreinish** [comment]
I would probably go ahead and bump the version string it so we know there were changes for terra 0.15.0 vs earlier releases. Even if nothing in the call path respects the version string yet starting to at least have a policy in terra for updating the version once per release will be something we can carry forward for when we get the schema out of terra and have things respect the version

**Comment by blakejohnson** [comment]
@mtreinish minor version bump?

File: qiskit/schemas/qobj_schema.json [hunk 1]
```diff
@@ -87,6 +87,22 @@
             ],
```
Comment (taalexander) [inline qiskit/schemas/qobj_schema.json:87]: Lets bump the version as per @mtreinish comment and than we should be good to go.

Comment (SooluThomas) [reply qiskit/schemas/qobj_schema.json:87]: Fixed in https://github.com/Qiskit/qiskit-terra/pull/4677/commits/aa84449b0258a5a817fefd056f19e69a382ba958

**Comment by mtreinish** [comment]
@blakejohnson I guess. We don't really have a versioning scheme (since the version has been `1.0.0` since the introduction of qobj), but we can use semver.

File: qiskit/schemas/qobj_schema.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "$schema": "http://json-schema.org/draft-04/schema#",
     "id": "http://www.qiskit.org/schemas/qobj_schema.json",
     "description": "OpenQuantum quantum object data structure for running experiments",
-    "version": "1.0.0",
+    "version": "1.1.0",
```
Comment (mtreinish) [inline qiskit/schemas/qobj_schema.json:5]: I was going to ask if you could update the outbound payload version (assuming we're sending `discriminator` and `kernel` in the qobj) so when there is proper schema version support the server know what we're sending. But it looks like that's already set to `1.1.0`:

https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/qobj/pulse_qobj.py#L515
https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/qobj/qasm_qobj.py#L408

Comment (blakejohnson) [reply qiskit/schemas/qobj_schema.json:5]: So do we need to bump again to `1.2.0`?

**Comment by taalexander** [comment]
If bumping to 1.2.0, then I believe we have to bump the Qobj objects as per @mtreinish's comment.

**Comment by blakejohnson** [comment]
@SooluThomas you also need to bump the schema version in the various tests in `test_qobj.py`.

**Comment by SooluThomas** [comment]
Yes, I tried bumping the schema version in `test_qobj.py` locally but I'm still getting the same error:
```
AssertionError: {'exp[164 chars]e': 0.0}, {'t0': 5, 'name': 'fc', 'ch': 'd0', [798 chars]: {}} != 
{'exp[164 chars]e': 0}, {'t0': 5, 'name': 'fc', 'ch': 'd0', 'p[789 chars]: {}}

Diff is 3083 characters long. Set self.maxDiff to None to see it.
```

**Pull request merged**